lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,12 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default),
+        ``input_to_hidden.input_shape[0]`` must be
```

Comment:
Oops, I only fixed it in one place, sorry.  The wording of naming `None` once is confusing to me.  If you want to change it or make other fixes, maybe it would be faster for you to make them then squash/merge?  This is a pretty small change.